mldsa/polyvec.h

```diff
@@ -356,7 +356,15 @@ __contract__(
  * Returns 0 if norm of all polynomials are strictly smaller than B <=
  *(MLDSA_Q-1)/8 and 1 otherwise.
  **************************************************/
-int polyveck_chknorm(const polyveck *v, int32_t B);
+int polyveck_chknorm(const polyveck *v, int32_t B)
+__contract__(
+  requires(memory_no_alias(v, sizeof(polyveck)))
+  requires(0 <= B && B <= (MLDSA_Q - 1) / 8)
+  requires(forall(k0, 0, MLDSA_K,
+                  array_bound(v->vec[k0].coeffs, 0, MLDSA_N,
+                              -REDUCE_RANGE_MAX, REDUCE_RANGE_MAX)))
+  ensures(return_value == 0 || return_value == 1)
```

Comment:
Quick question: Did you intentionally not add the 
```
ensures((return_value == 0) == forall(k0, 0, MLDSA_K, 
    array_abs_bound(v->vec[k0].coeffs, 0, MLDSA_N, B)))
```
?

Reply:
Yeah... not sure it will be needed in sign.c so I didn't put it in.

Reply:
Just checked. I think we don't need it.